cou/apps/base.py

```diff
@@ -631,12 +646,28 @@ def _get_upgrade_charm_step(self, target: OpenStackRelease) -> UpgradeStep:
         :return: Step for upgrading the charm.
         :rtype: UpgradeStep
         """
-        if self.channel != self.target_channel(target):
+        channel = self.expected_current_channel(target) if self.need_crossgrade else self.channel
+
+        if (
+            self.current_channel_os_release <= self.current_os_release
+            and channel != self.target_channel(target)
+        ):
             return UpgradeStep(
-                description=f"Upgrade '{self.name}' to the new channel: "
+                description=f"Upgrade '{self.name}' from '{channel}' to the new channel: "
                 f"'{self.target_channel(target)}'",
                 coro=self.model.upgrade_charm(self.name, self.target_channel(target)),
             )
+
+        if (
+            self.current_channel_os_release > self.current_os_release
+            and channel != self.target_channel(target)
+        ):
+            description, coro = self._get_change_channel_possible_downgrade_step(
+                target, self.target_channel(target)
+            )
+            return UpgradeStep(description=description, coro=coro)
```

Comment:
Why once this is PreUpgradeStep and now UpgradeStep?

Reply:
Because a downgrade can happen on a pre-upgrade step e.g from latest/stable to ussuri/stable and also as a upgrade-step, e.g from `quincy/stable` to `octopus/stable`

Reply:
In the end we had an agreement to raise an error on the case of the `UpgradeStep`